pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -176,6 +184,35 @@ func (r *ReconcileToolchainStatus) aggregateAndUpdateStatus(reqLogger logr.Logge
 
 	// if any components were not ready then set the overall status to not ready
 	if len(unreadyComponents) > 0 {
+		// If the current ToolchainStatus:
+		// a) Is currently not ready,
+		// b) has not been ready for longer than the configured threshold, and
+		// c) no notification has been already sent, then
+		// send a notification to the admin mailing list
+		c, found := condition.FindConditionByType(toolchainStatus.Status.Conditions, toolchainv1alpha1.ConditionReady)
+		if found && c.Status == corev1.ConditionFalse {
+			threshold := time.Now().Add(-minutesAfterUnready * time.Minute)
+			if c.LastUpdatedTime.Before(&metav1.Time{threshold}) {
+				c, found = condition.FindConditionByType(toolchainStatus.Status.Conditions,
+					toolchainv1alpha1.ToolchainStatusUnreadyNotificationCreated)
+				if !found {
+					if err := r.sendToolchainStatusUnreadyNotification(reqLogger, toolchainStatus); err != nil {
+						reqLogger.Error(err, "Failed to create toolchain status unready notification")
+
+						// set the failed to create notification status condition
+						return r.wrapErrorWithStatusUpdate(reqLogger, toolchainStatus,
+							r.setStatusUnreadyNotificationCreationFailed, err,
+							"Failed to create user deactivation notification")
+					}
+
+					if err := r.setStatusToolchainStatusUnreadyNotificationCreated(toolchainStatus); err != nil {
+						reqLogger.Error(err, "Failed to update notification created status")
+						return err
+					}
+				}
+			}
+		}
+
```

Comment:
This is cosmetic but... we could move it to a separate function to improve code readability. But feel free to keep it as it is if you prefer it this way.

Reply:
Sure thing, moved.